OutOfSchool/OutOfSchool.WebApi/Controllers/V1/WorkshopController.cs

```diff
@@ -218,17 +220,54 @@ public async Task<IActionResult> Update(WorkshopDTO dto)
 
             var userHasRights = await this.IsUserProvidersOwnerOrAdmin(dto.ProviderId, dto.Id).ConfigureAwait(false);
 
-            if (dto.ProviderOwnership == OwnershipType.Common || dto.ProviderOwnership == OwnershipType.State)
+            if (!userHasRights)
             {
-                dto.Status = WorkshopStatus.Open;
+                return StatusCode(403, "Forbidden to update workshops, which are not related to you");
             }
 
+            return Ok(await combinedWorkshopService.Update(dto).ConfigureAwait(false));
+        }
+
+        /// <summary>
+        /// Update status field for workshop entity.
+        /// </summary>
+        /// <param name="request">Workshop id and status to update.</param>
+        /// <returns>Updated <see cref="WorkshopStatusDto"/>.</returns>
+        /// <response code="200">Entity was updated and returned.</response>
+        /// <response code="400">If the model is invalid, some properties are not set etc.</response>
+        /// <response code="401">If the user is not authorized.</response>
+        /// <response code="403">If the user has no rights to use this method, or sets some properties that are forbidden to change.</response>
+        /// <response code="500">If any server error occures.</response>
+        [HasPermission(Permissions.WorkshopEdit)]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(WorkshopStatusDto))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpPut]
+        public async Task<IActionResult> UpdateStatus([FromBody] WorkshopStatusDto request)
+        {
+            var workshop = await combinedWorkshopService.GetById(request.WorkshopId).ConfigureAwait(false);
```

Comment:
Please, validate parameter 'request' is non-null before using it.
Please review also other sonar issues.

Reply:
There is no need for such a request validation, this check is performed by the framework.